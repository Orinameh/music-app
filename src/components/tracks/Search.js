import React, { Component } from 'react'
import axios from 'axios' 
import { Consumer } from '../../context'

class Search extends Component {

	state = {
		trackTitle: ''
	}

	onChange = e => {
		this.setState({ trackTitle: e.target.value})
	}

	findTrack = (e, dispatch) => {
		e.preventDefault()
		axios.get(`https://cors-anywhere.herokuapp.com/https://api.musixmatch.com/ws/1.1/track.search?q_track=${this.state.trackTitle}&page_size=10&s_track_rating=desc&apikey=${process.env.REACT_APP_API_KEY}`)
		.then(res => {
			dispatch({
				type: 'SEARCH_TRACKS',
				payload: res.data.message.body.track_list
			})
			this.setState({ trackTitle: ''})
		})
	}
	render() {
		return (
			<Consumer>
				{
					value => {
						const { dispatch } = value
						return (
							<div className="card cardy-body mb-4 p-4">
								<h1 className="display-4 text-center">
									 <i className="fas fa-music" aria-hidden="true"></i> Search a Song
								</h1>
								<p className="lead text-center">Get the lyrics for any song</p>

								<form onSubmit={e => this.findTrack(e, dispatch)}>
									<div className="form-group">
										<input 
											type="text" 
											className="form-control form-control-lg"
											name="trackTitle" 
											placeholder="Song title"
											value={this.state.trackTitle}
											onChange={this.onChange}/>
									</div>
									<button type="submit" className="btn btn-primary btn-lg btn-block mb-5">
										Get Track Lyrics
									</button>
								</form>
							</div>
						)
					}
				}
			</Consumer>
		)
	}
}

export default Search
