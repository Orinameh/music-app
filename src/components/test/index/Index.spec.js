import React from 'react';
import { shallow, mount } from 'enzyme';
import Index from '../../layout/Index'
import Search from '../../tracks/Search'
import Tracks from '../../tracks/Tracks';

describe('Index', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = shallow(<Index />)
	})

	it('should render correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should render a <Fragment />', () => {
		expect(wrapper.find('Fragment').length).toEqual(1)
	});

	it('should render the Search and Tracks Component', () => {
		expect(wrapper.containsAllMatchingElements([
			<Search />,
			<Tracks />
		])).toEqual(true)
	})
	
})
