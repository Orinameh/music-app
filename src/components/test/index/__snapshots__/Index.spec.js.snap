// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Index />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Search />,
        <Tracks />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Search />,
          <Tracks />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
